examples/dynamics/braking_path_after_collision.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+
+from sympy import solve, Symbol, Eq
+from symplyphysics import Quantity, convert_to, units, print_expression
+from symplyphysics.definitions import (
+    momentum_is_mass_times_velocity as momentum_def,
+)
+from symplyphysics.laws.conservation import (
+    mechanical_energy_after_equals_to_mechanical_energy_before as energy_conservation_law,
+    momentum_after_collision_equals_to_momentum_before as momentum_conservation_law
+)
+from symplyphysics.laws.dynamics import (
+    kinetic_energy_from_mass_and_velocity as kinetic_energy_def,
+    potential_energy_from_mass_and_height as potential_energy_def,
+    friction_force_from_normal_force as friction_law,
+    braking_path as braking_path_law,
+)
+
+# Description
+## Block 1 of mass m1 slides from rest along a frictionless ramp from height h = 2.50 m and then
+## collides with stationary block 2, which has mass m2 = 2.00*m1. After the collision, block 2
+## slides into a region where the coefficient of kinetic friction is mu = 0.500 and comes to a
+## stop in distance d within that region. What is the value of distance d if the collision was
+## (a) elastic and (b) completely inelastic?
+
+mass_1 = Symbol("mass_1")
+mass_2 = 2 * mass_1
+height = Symbol("height")
+friction_coefficient = Symbol("friction_coefficient")
+
+values = {
+    height: Quantity(2.50 * units.meter),
+    friction_coefficient: Quantity(0.500),
+    units.acceleration_due_to_gravity: Quantity(9.81 * units.meter / units.second**2),
+}
+
+speed_before_1 = Symbol("speed_before_1")
+speed_after_1 = Symbol("speed_after_1")
+speed_after_2 = Symbol("speed_after_2")
+
+# Find speed of block 1 just before the collision
+
+energy_before_1 = potential_energy_def.law.rhs.subs({
+    potential_energy_def.height: height,
+    potential_energy_def.body_mass: mass_1,
+    potential_energy_def.free_fall_acceleration: units.acceleration_due_to_gravity,
+})
+
+energy_after_1 = kinetic_energy_def.law.rhs.subs({
+    kinetic_energy_def.body_mass: mass_1,
+    kinetic_energy_def.body_velocity: speed_before_1,
+})
+
+energy_conservation_eqn_1 = energy_conservation_law.law.subs({
```

Comment:
Energy conservation acts on mechanical energy. An mechanical energy is a sum of kinetic and potential energy. So it might be more correct to use https://github.com/blackyblack/symplyphysics/blob/master/symplyphysics/definitions/mechanical_energy_is_kinetic_and_potential.py to obtain mechanical energy before passing it to energy conservation equation.

Reply:
I agree, will add